New Website Theme

waht opened the pull request:
# Purpose

New website theme with updated logo, new fonts, colors and minor changes to the start page.
Here is a preview:
![theme](https://user-images.githubusercontent.com/2136968/35013181-d85f0d4c-fb0c-11e7-8e3f-4a9da2795840.png)

# Checklist

- [x] commit messages are fine (with references to issues)
- [x] I did local tests and everything went fine
- [x] release notes are updated (doc/news/_preparation_next_release.md)

@markus2330 please review my pull request

markus2330 commented on doc/news/_preparation_next_release.md:
```diff
@@ -34,22 +34,25 @@ You can also read the news [on our website](https://www.libelektra.org/news/0.8.
 ## Highlights
 
-- New Logo
+- New Logo and Website Theme
```
At the moment it even looks like we can make an extra highlight for this.

markus2330 commented on doc/news/_preparation_next_release.md:
```diff
@@ -34,22 +34,25 @@ You can also read the news [on our website](https://www.libelektra.org/news/0.8.
 Thanks to Philipp Frei
 
+We also gave the website a new look. It has the colors from the logo, new
+fonts for better readability and the start page structure was improved.
```
Which font? Please add link to Website here.

markus2330 commented on src/tools/rest-frontend/resources/assets/html/index.html.in:
```diff
@@ -22,6 +22,9 @@
         <link rel="alternate" type="application/atom+xml" href="news/feed.rss" title="Elektra RSS Feed">
 
+        <!-- Fonts -->
+        <link href="https://fonts.googleapis.com/css?family=Lato|Libre+Franklin" rel="stylesheet">
```
Please do not use cross-domain requests but copy the font into the repo (As done for other fonts already.).

Maybe some fonts are now obsolete and can be removed?

waht replied:
Sure, this was intended for testing the fonts. I would convert and integtrate the fonts when we agreed upon them.

markus2330 commented on src/tools/rest-frontend/resources/assets/skin/bootstrap/list-group.less:
```diff
@@ -0,0 +1,6 @@
+    background-image: none;
+    box-shadow: none;
+  }
+}
```
Every text file should end with a new-line.

markus2330 commented on src/tools/rest-frontend/resources/assets/skin/bootstrap/navbar.less:
```diff
@@ -0,0 +1,12 @@
+  }
+ }
+ 
+ 
```
Space?

markus2330 commented:
jenkins build homepage please

markus2330 commented:
The new website is online, I love it.

I noticed that the colors of the original buttons sometimes do not fit well, e.g. at https://www.libelektra.org/conversion but the buttons might be changed by @vanessaNWK anyway.

The font change is really minor, is it worth the trouble? Is it now the same font as in the logo?

waht commented:
Thank you! Seems to work great!

Can you be more specific about the buttons at the conversion page? Is it the blue "conversion" button or the red "report" button?
![conversion](https://user-images.githubusercontent.com/2136968/35418086-94ec7614-0230-11e8-95e6-812fd7d6f25c.png)

No its not the font from the logo (that did not look good at all). Granted, the difference is not that big but it improves readability (at least on my system) and helps to build a unique look for the website.
I created some screenshots with and without the fonts for comparison:

- [home with fonts](https://user-images.githubusercontent.com/2136968/35418248-89dc6fa8-0231-11e8-89e2-2216b81ae536.png)
- [home without fonts](https://user-images.githubusercontent.com/2136968/35418249-89f5fb4e-0231-11e8-8ae8-1a7a9fbb9748.png)
- [docs with fonts](https://user-images.githubusercontent.com/2136968/35418250-8a123dea-0231-11e8-984a-52bd720f1fd5.png)
- [docs without fonts](https://user-images.githubusercontent.com/2136968/35418251-8a308f48-0231-11e8-9af3-678a8521b7ac.png)

markus2330 commented:
> Can you be more specific about the buttons at the conversion page? Is it the blue "conversion" button or the red "report" button?

If my eyes do not betray me both colors do not match with the color scheme. It is really minor, though. It is definitely not a necessity to change this. (even if they are mismatching colors, which I did not check with a tool nor a side-by-side comparison).

> home without fonts

Okay, my "home without fonts" looks completely different, seems to be an issue of which fonts are installed. Let us pick up this discussion in the meeting today.

markus2330 commented:
jenkins build homepage please

markus2330 commented:
I have triggered a rebuild so that we can see the new website in the meeting today.

Btw. the RSS symbol could also be in Elektra's orange. (Again, really minor)

waht commented:
jenkins build homepage please

markus2330 commented:
Looks great now! There are some formatting issues (spaces in CSS files) and could you please add SPDX-License-Identifier: SIL (see [SPDX](https://spdx.org/licenses/) and [reuse](https://reuse.software/practices/2.0/)). SIL seems to be designed to be distributed in ways like this but maintainers need to know that we do so. (They are copyleft and not BSD.)

Did you find any font not in use anymore?

waht commented:
Thank you! I see, I replaced the spaces with tabs in the style files. 

About the licensing: The font files are binary so I could not add the SPDX identifier - I opted for an [DEP-5/copyright](https://www.debian.org/doc/packaging-manuals/copyright-format/1.0) file to state the license for all font files at once. I also searched for further copyright notes in files and included those along with their licenses as well. 
SPDX would also recommend adding all license texts in a LICENSES directory, but I haven't done that yet.

I removed the symbol font "font-awesome" from our source in favor of an npm-package.

markus2330 commented:
Thank you for taking care of the licensing. In the current situation we would get a merge conflict when we merge master into our debian branch which already has a debian/copyright file. I am not aware of a standard of how upstream can provide a debian/copyright file. I think the easiest way would be that you merge it into our debian branch in the already existing [debian/copyright](https://github.com/ElektraInitiative/libelektra/blob/debian/debian/copyright) file. (A separate PR is needed for this.)

Is the npm package of font-awesome reliable? We already had quite often problems of npm packages that suddenly break. It also seems weird that some fonts are included and other are downloaded as package.

Otherwise I cannot wait to get this merged, the website improvement is fabulous.

markus2330 commented:
jenkins build homepage please

waht commented:
I understand - the package [seems quite fine](https://packagequality.com/#?package=font-awesome), but I added font-awesome back to keep things consistent. I'm creating the PR for debian right now.

waht commented:
jenkins build homepage please

markus2330 commented:
For me even the website packagequality.com does not look trustworthy :-)

If you think its better this way, we can try it. But please do this migration in a separate commit so that we can revert it if the package causes trouble.

waht commented:
My intuition is that the package is fine. It has versions along regular releases of the font-awesome project and the publisher is a (one-time though) contributor to the project.

But as long the other fonts are not available via npm it's not worth the trouble, and things are better if they are consistent.

markus2330 commented on src/tools/rest-frontend/package.json:
```diff
@@ -36,6 +36,7 @@
         "bootstrap": "^3.3.7",
         "connect-modrewrite": "^0.9.0",
         "fs-extra": "^0.30.0",
+        "font-awesome": "^4.7.0",
```
Why do we need this line if we also have the fonts included in the repo?

waht replied:
Yeah - we don't :) I'll remove it.

markus2330 commented:
Thank you! I'll trigger the website build from master.
